backend/internal/controllers/book_test.go

```diff
@@ -50,3 +52,155 @@ func TestGetNullBook(t *testing.T) {
 
 	require.Equal(t, "book not found", response)
 }
+
+// Test to just create the book
+func TestJustCreateBook(t *testing.T) {
+	body := `{"title": "Lonely Book", "author":"Lonely Man", "image":null}`
+	testutils.MakeHttpRequest("POST", ts.URL+"/books", body, http.StatusCreated,
+		nil, t)
+
+	var response []models.Book
+
+	// get req should not cause an error if book exists without any languages
+	testutils.MakeHttpRequest("GET", ts.URL+"/books", "", 200, &response, t)
+
+	require.Len(t, response, 6)
+}
+
+// Test creating a book and its contents together
+func TestCreateBookandBookDetails(t *testing.T) {
+	body := `{"title": "Harry Potter", "author": "JK Rowling", "image":null}`
+	var createdBook models.Book
+	testutils.MakeHttpRequest("POST", ts.URL+"/books", body, http.StatusCreated,
+		&createdBook, t)
+
+	require.Equal(t, "Harry Potter", createdBook.Title)
+	require.Equal(t, "JK Rowling", createdBook.Author)
+	require.Equal(t, []string{}, createdBook.Languages)
+
+	body = `{"language": "en", 
+	"read": {"video": null, "body":"hp_rb"}, 
+	"explore": {"video": null, "body":"hp_eb"},  
+	"learn": {"video": null, "body":"hp_lb"}}`
+	var createdBookDetails models.BookDetails
+	testutils.MakeHttpRequest("POST", ts.URL+"/books/"+createdBook.ID, body, http.StatusCreated,
+		&createdBookDetails, t)
+
+	require.Equal(t, "hp_rb", createdBookDetails.Read.Body)
+	require.Equal(t, "hp_eb", createdBookDetails.Explore.Body)
+}
+
+// Make sure foreign key constraint is working
+func TestBookDetailsForeignKey(t *testing.T) {
+	body := `{"language": "en", 
+	"read": {"video": null, "body":"bad_rb"}, 
+	"explore": {"video": null, "body":"bad_eb"},  
+	"learn": {"video": null, "body":"bad_lb"}}`
+
+	testutils.MakeHttpRequest("POST", ts.URL+"/books/nonexistant", body, http.StatusInternalServerError,
```

Comment:
> `http.StatusInternalServerError`

Oh damn, actually it should be very difficult for an API client to intentionally cause a 5xx error. We should have better error handling in any PATCH/POST/DELETE routes that take an ID in the URL, in order to throw a 404 if that ID doesn't exist. Can you please add this? It should be pretty straightforward, just `GetBookByID/GetBookDetailsByID(id)` and return a 404 if the result is nil.